File: src/java/com/portalpostal/dao/handler/ContaHandler.java
package com.portalpostal.dao.handler;

import com.portalpostal.model.CartaoCredito;
import com.portalpostal.model.Conta;
import com.portalpostal.model.ContaCorrente;
import com.portalpostal.model.dd.TipoConta;
import com.portalpostal.model.dd.TipoStatusConta;
import java.sql.ResultSet;
import java.sql.SQLException;
import org.sql2o.ResultSetHandler;

public class ContaHandler extends GenericHandler implements ResultSetHandler<Conta> {
        
    public ContaHandler() {
        super("conta");
    }
    
    public ContaHandler(String table) {
        super(table);
    }

    public Conta handle(ResultSet result) throws SQLException {
        Conta conta = new Conta();
        conta.setIdConta(getInt(result, "idConta"));
        conta.setNome(getString(result, "nome"));        
        conta.setTipo(TipoConta.values()[getInt(result, "tipo")]);              
        conta.setStatus(TipoStatusConta.values()[getInt(result, "status")]);
        conta.setDataAbertura(getDate(result, "dataAbertura"));
        conta.setValorSaldoAbertura(getDouble(result, "valorSaldoAbertura"));
        conta.setContaCorrente(getContaCorrente(result));
        conta.setCartaoCredito(getCartaoCredito(result));
        return conta;
    }
    
    private ContaCorrente getContaCorrente(ResultSet result) throws SQLException {
        if(!existColumn(result, "conta_corrente.idContaCorrente")) return null;
        return new ContaCorrenteHandler().handle(result); 
    }
    
    private CartaoCredito getCartaoCredito(ResultSet result) throws SQLException {
        if(!existColumn(result, "cartao_credito.idCartaoCredito")) return null;  
        return new CartaoCreditoHandler().handle(result);
    }
    
}

File: web/app/js/handler/contaHandler.js
'use strict';

app.factory('ContaHandler', function() {
    
    var handleList = function(contaList) { 
        return _.map(contaList, function(conta) {
            return handle(conta);
        });
    };
    
    var handle = function(conta) {  
        var contaHandle = {};
        contaHandle.idConta = getId(conta); 
        contaHandle.contaCorrente = getContaCorrente(conta);
        contaHandle.cartaoCredito = getCartaoCredito(conta);
        contaHandle.nome = getNome(conta);
        contaHandle.tipo = getTipo(conta);
        contaHandle.status = getStatus(conta);
        contaHandle.dataAbertura = getDataAbertura(conta);
        contaHandle.valorSaldoAbertura = getValorSaldoAbertura(conta);
        contaHandle.saldo = getSaldo(conta);
        contaHandle.codigoIntegracao = getCodigoIntegracao(conta);
        contaHandle.visivel = getVisivel(conta);
        return contaHandle;
    };
    
    var getId = function(conta) {
        return conta.idConta || null;
    };
    
    var getContaCorrente = function(conta) {
        if(!conta.contaCorrente) return conta.contaCorrente;
        return { idContaCorrente: conta.contaCorrente.idContaCorrente };        
    };
    
    var getCartaoCredito = function(conta) {
        if(!conta.cartaoCredito) return conta.cartaoCredito;
        return { idCartaoCredito: conta.cartaoCredito.idCartaoCredito };        
    };
    
    var getNome = function(conta) {
        return conta.nome || null; 
    };
    
    var getTipo = function(conta) {
        if(!conta.tipo) return conta.tipo;
        return conta.tipo.id; 
    };
    
    var getStatus = function(conta) {
        if(!conta.status) return conta.status;
        return conta.status.id; 
    };
    
    var getDataAbertura = function(conta) {
        return conta.dataAbertura || null; 
    };
    
    var getValorSaldoAbertura = function(conta) {
        return conta.valorSaldoAbertura || null; 
    };
    
    var getSaldo = function(conta) {
        return conta.saldo || null; 
    };
    
    var getCodigoIntegracao = function(conta) {
        return conta.codigoIntegracao || 1; 
    };
    
    var getVisivel = function(conta) {
        return conta.visivel; 
    };

    return {
        handle: handle,
        handleList: handleList
    }

});